ted_sws/notice_metadata_processor/services/notice_eligibility.py

```diff
@@ -18,22 +42,29 @@ def check_package(mapping_suite: MappingSuite, notice_metadata: NormalisedMetada
     :param mapping_suite:
     :return:
     """
+
     constraints = mapping_suite.metadata_constraints.constraints
 
     eform_subtype = notice_metadata.eforms_subtype
     notice_publication_date = datetime.datetime.fromisoformat(notice_metadata.publication_date)
-    notice_xsd_version = notice_metadata.xsd_version
 
-    end_date = constraints[END_DATE_KEY][0] if constraints[END_DATE_KEY] else datetime.datetime.now().isoformat()
-    constraint_start_date = datetime.datetime.fromisoformat(constraints[START_DATE_KEY][0])
-    constraint_end_date = datetime.datetime.fromisoformat(end_date)
-    constraint_min_xsd_version = constraints[MIN_XSD_VERSION_KEY][0]
-    constraint_max_xsd_version = constraints[MAX_XSD_VERSION_KEY][0]
+    if notice_metadata.is_eform:
+        notice_xsd_version = notice_metadata.eform_sdk_version
+        eforms_sdk_version = notice_xsd_version.rsplit('-', 1)[1]
+        constraint_version_range = [format_version_with_zero_patch(version) for version in
+                                    constraints[EFORMS_SDK_VERSIONS_KEY]]
+        in_version_range = format_version_with_zero_patch(eforms_sdk_version) in constraint_version_range
+    else:
+        notice_xsd_version = notice_metadata.xsd_version
+        constraint_min_xsd_version = constraints[MIN_XSD_VERSION_KEY][0]
+        constraint_max_xsd_version = constraints[MAX_XSD_VERSION_KEY][0]
+        in_version_range = constraint_min_xsd_version <= notice_xsd_version <= constraint_max_xsd_version
+
+    in_date_range = is_date_in_range(publication_date=notice_publication_date,
+                                     constraint_start_date_value=constraints[START_DATE_KEY],
+                                     constraint_end_date_value=constraints[END_DATE_KEY])
     eform_subtype_constraint_values = [str(eforms_subtype_value) for eforms_subtype_value in
                                        constraints[E_FORMS_SUBTYPE_KEY]]
-
-    in_date_range = constraint_start_date <= notice_publication_date <= constraint_end_date
-    in_version_range = constraint_min_xsd_version <= notice_xsd_version <= constraint_max_xsd_version
     covered_eform_type = eform_subtype in eform_subtype_constraint_values
 
     return True if in_date_range and in_version_range and covered_eform_type else False
```

Comment:
replace:
`return True if in_date_range and in_version_range and covered_eform_type else False`
with:
`return in_date_range and in_version_range and covered_eform_type`